### PR TITLE
treesitter: unknown predicates always match

### PR DESCRIPTION
**Pull request by vigoux**
This PR changes slightly the behaviour of treesitter integration so that unknown predicates always match.
This allows for plugins like `nvim-treesitter` to add and test predicates before doing PRs so that they are included here.

Also change the behaviour of the `match?` to not return unconditionnaly.

### CODE REVIEW

File: runtime/lua/vim/treesitter.lua [hunk 1]
```diff
@@ -184,9 +184,9 @@ function Query:match_preds(match, pattern, bufnr)
       if start_row ~= end_row then
         return false
       end
-      return regexes[i]:match_line(bufnr, start_row, start_col, end_col)
-    else
-      return false
+      if not regexes[i]:match_line(bufnr, start_row, start_col, end_col) then
+        return false
+      end
```
Comment (justinmk) [inline runtime/lua/vim/treesitter.lua:189]: would prefer to keep the explicit `else`, perhaps with a hint

    else  -- Unknown predicate always matches. #12173
      return true

Comment (vigoux) [reply runtime/lua/vim/treesitter.lua:189]: The thing is that we don't want to return here to go through all the predicates in this match. This returning `true` only at the very end of the function.

Comment (justinmk) [reply runtime/lua/vim/treesitter.lua:189]: ok that's definitely worth calling out in a comment :)

Comment (vigoux) [reply runtime/lua/vim/treesitter.lua:192]: You are right, it should be solved in the last push.

File: runtime/lua/vim/treesitter.lua [hunk 2]
```diff
@@ -159,6 +159,9 @@ function Query:match_preds(match, pattern, bufnr)
   end
   local regexes = self.regexes[pattern]
   for i, pred in pairs(preds) do
+    -- Here we only want to return if a predicate DOES NOT match, and
+    -- continue on the other case. This way unknown predicates will not be considered,
```
Comment (justinmk) [inline runtime/lua/vim/treesitter.lua:163]:
```suggestion
    -- continue on the other case. This way unknown predicates will not be considered,
```

**Pull request merged**